lib/plugins/aws/package/compile/events/cloudFront/index.js

```diff
@@ -177,25 +178,36 @@ class AwsCompileCloudFrontEvents {
                 ? event.cloudFront.pathPattern
                 : undefined;
 
-            let origin = createOrigin(functionName, event.cloudFront.origin, this.provider.naming);
-
-            const existingOrigin = _.find(
-              origins,
-              o => o.DomainName === origin.DomainName && o.OriginPath === origin.OriginPath
-            );
+            let origin = createOrigin(event.cloudFront.origin, this.provider.naming);
+            const existingOrigin = origins.find(o => o.Id === origin.Id);
 
             if (!existingOrigin) {
               origins.push(origin);
             } else {
               origin = extendDeep(existingOrigin, origin);
             }
 
+            if (event.cloudFront.defaultOrigin === true) {
+              if (defaultOrigin && defaultOrigin !== origin) {
+                throw new ServerlessError(
+                  'Found more than one cloudfront event with "defaultOrigin" defined'
+                );
+              }
+              defaultOrigin = origin;
+            }
+
             let behavior = {
               ViewerProtocolPolicy: 'allow-all',
-              TargetOriginId: origin.Id,
               ForwardedValues: {
                 QueryString: false,
               },
+            };
+            if (_.isObject(event.cloudFront.behavior)) {
```

Comment:
If I understand correct, object representation it's only one we accept (there's no string version).

In that case, it's better written as `if (event.cloudFront.behavior)`